airbyte-ci/connectors/connectors_qa/connectors_qa/checks/metadata.py

```diff
@@ -0,0 +1,160 @@
+
+    PYTHON_LANGUAGE_TAG = "language:python"
+    JAVA_LANGUAGE_TAG = "language:java"
+
+    def _run(self, connector: Connector) -> CheckResult:
+        current_language_tags = [t for t in connector.metadata.get("tags", []) if t.startswith("language:")]
+        if not current_language_tags:
+            return self.fail(
+                connector=connector,
+                message="Language tag is missing in the metadata file",
+            )
+        if len(current_language_tags) > 1:
+            return self.fail(
+                connector=connector,
+                message=f"Multiple language tags found in the metadata file: {current_language_tags}",
+            )
+        current_language_tag = current_language_tags[0]
+
+        if (connector.code_directory / consts.SETUP_PY_FILE_NAME).exists() or (
+            connector.code_directory / consts.PYPROJECT_FILE_NAME
+        ).exists():
+            expected_language = self.PYTHON_LANGUAGE_TAG
+        elif (connector.code_directory / consts.GRADLE_FILE_NAME).exists():
+            expected_language = self.JAVA_LANGUAGE_TAG
+        else:
+            return self.fail(
+                connector=connector,
+                message="Could not infer the language tag from the connector directory",
+            )
+        if current_language_tag != expected_language:
+            return self.fail(
+                connector=connector,
+                message=f"Expected language tag '{expected_language}' in the {consts.METADATA_FILE_NAME} file, but found '{current_language_tag}'",
+            )
+        return self.pass_(
+            connector=connector,
+            message=f"Language tag {expected_language} is present in the metadata file",
+        )
+
+class ValidateBreakingChangesDeadlines(MetadataCheck):
+    """
+    Verify that _if_ the the most recent connector version has a breaking change,
+    it's deadline is at least a week in the future.
+    """
+    name = "Breaking change opt-in should be a week in the future"
+    description = f"If the connector version has a breaking change, the deadline field must be set to at least a week in the future."
+
+    def _run(self, connector: Connector) -> CheckResult:
+
+        # fetch the current branch version of the connector first.
+        # we'll try and see if there are any breaking changes associated
+        # with it next.
+        current_version = connector.version
+        if current_version is None:
+            return self.fail(
+                connector=connector,
+                message="Can't verify breaking changes deadline: connector version is not defined.",
+            )
+
+        breaking_changes = connector.metadata['releases']['breakingChanges']
```

Comment:
This might raise a `KeyError` if the metadata does not have a `releases` or `breakingchanges`
```suggestion
        from pydash import get
        breaking_changes = get(connector.metadata, "releases.breakingChanges")
```